// crm-prototypes/dist/formatters/response.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Response = void 0;
// @ts-ignore
const common_1 = require("@nestjs/common");
const messages_1 = require("./messages");
class Response {
    static findSuccess([result, count]) {
        return {
            messages: [messages_1.Messages.getSuccess],
            data: {
                result,
                total: count,
            },
        };
    }
    static createSuccess(result) {
        return {
            messages: [messages_1.Messages.createSuccess],
            data: {
                result,
                total: 1,
            },
        };
    }
    static createBulkSuccess(result) {
        return {
            messages: [messages_1.Messages.createSuccess],
            data: {
                result,
                total: result.length,
            },
        };
    }
    static updateSuccess(result) {
        return {
            messages: [messages_1.Messages.updateSuccess],
            data: {
                result,
                total: 1,
            },
        };
    }
    static deleteSuccess(result) {
        return {
            messages: [messages_1.Messages.deleteSuccess],
            data: {
                result: null,
                total: result,
            },
        };
    }
    static badRequestThrow(message) {
        throw new common_1.BadRequestException({
            messages: Array.isArray(message) ? message : [message],
            data: {
                result: null,
                total: 0,
            },
        });
    }
}
exports.Response = Response;
